code/game/area/areas.dm

```diff
@@ -238,6 +156,13 @@
 		party = 0
 		mouse_opacity = 0
 		updateicon()
+		for(var/obj/machinery/door/firedoor/D in src)
+			if(!D.blocked)
+				if(D.operating)
+					D.nextstate = OPEN
+				else if(D.density)
+					spawn(0)
+					D.open()

```

Comment:
Not sure what's up with that diff - the code itself doesn't actually have proper tabbing. Must be an issue with Github.